File: Python/pcapng.py
import conv

def read(fl):
	while True:
		h = fl.read(4)
		if not h:
			break
		type = conv.bytes_int_le(h)
		h = fl.read(4)
		len = conv.bytes_int_le(h)
		b = fl.read(len-12)
		fl.read(4)
		if type==6:
			r = {}
			sz = conv.bytes_int_le(b[12:16])
			r['size_lost'] = conv.bytes_int_le(b[16:20]) - sz
			r['frame'] = b[20:20+sz]
			yield r

File: Go/fls/parsefile/pcapng.go
package parsefile

import "encoding/binary"

type RDpcapng struct {
	readerFC
}

func (r *RDpcapng) Init(fl string){
	r.open(fl)
}

func (r *RDpcapng) Read() []byte {
	if r.fl == nil {
		return nil
	}
	tmp := make([]byte, 4)
	for {
		_, err := r.fl.Read(tmp)
		if err != nil {
			break
		}
		k := binary.LittleEndian.Uint32(tmp)
		_, err = r.fl.Read(tmp)
		if err != nil {
			break
		}
		l := binary.LittleEndian.Uint32(tmp)
		frame := make([]byte, l-12)
		_, err = r.fl.Read(frame)
		if err != nil {
			break
		}
		_, err = r.fl.Read(tmp)
		if err != nil {
			break
		}
		if k == 6 {
			sz := binary.LittleEndian.Uint32(frame[16:20])
			return frame[20:20+sz]
		}
	}
	r.Close()
	return nil
}
